pkg/cmd/cli/cmd/set/env.go

```diff
@@ -176,6 +177,20 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 
 	skipped := 0
 	for _, info := range infos {
+		ok := f.UpdateBuildStrategyForObject(info.Object, func(spec *buildapi.BuildStrategy) {
```

Comment:
This needs to be abstracted more - I'd like to see a method "EnvironmentForObject" that allows you to mutate.